saleor/graphql/page/dataloaders.py

```diff
@@ -1,10 +1,227 @@
-from ...page.models import Page
+from collections import defaultdict
+
+from promise import Promise
+
+from ...attribute.models import AssignedPageAttribute, AttributePage, AttributeValue
+from ...core.permissions import PagePermissions
+from ...page.models import Page, PageType
+from ..attribute.dataloaders import AttributesByAttributeId, AttributeValueByIdLoader
 from ..core.dataloaders import DataLoader
+from ..utils import get_user_or_app_from_context
 
 
 class PageByIdLoader(DataLoader):
     context_key = "page_by_id"
 
     def batch_load(self, keys):
-        pages = Page.objects.visible_to_user(self.user).in_bulk(keys)
+        requestor = get_user_or_app_from_context(self.context)
+        pages = Page.objects.visible_to_user(requestor).in_bulk(keys)
```

Comment:
In mulitchannel we decided to move visibility checks outside dataloaders, because in many places we need to pass channel to check if the object is visible. 
As far as I know, we have plans to move pages to multichannel.

Reply:
And it's harder to reuse dataloader if them doesn't return all objects.